Tasks/AzureFunctionOnKubernetesV1/src/clusterconnection.ts

```diff
@@ -0,0 +1,167 @@
+"use strict";
+
+import * as fs from "fs";
+import * as path from "path";
+import * as tl from "azure-pipelines-task-lib/task";
+import * as tr from "azure-pipelines-task-lib/toolrunner";
+import * as utils from "./utils/utilities";
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+
+export default class ClusterConnection {
+    private kubectlPath: string;
+    private kubeconfigFile: string;
+    private userDir: string;
+
+    constructor(existingKubeConfigPath?: string) {
+        this.kubectlPath = tl.which("kubectl", false);
+        this.userDir = utils.getNewUserDirPath();
+        if (existingKubeConfigPath) {
+            this.kubeconfigFile = existingKubeConfigPath;
+        }
+    }
+
+    private loadClusterType(connectionType: string): any {
```

Comment:
The require refers files added in this pull request as well.

I copied this implementation over from the `KubernetesV1` task; I think that the `any` is here because they decided not to import both implementations at the same time, and they don't have a common interface declared somewhere, so there's no common type that this returns. I agree that this is somewhat suboptimal, but I decided not to refactor the code I was copying from KubernetesV1 too much, because there are a lot of things I would like to refactor about it :D